### ISSUE TITLE
Type check failing leads to inability to change pipeline

### ISSUE THREAD

**Issue by jluethi** [TissueMAPS/TissueMAPS#53]
Working on the most recent release candidate for 0.3.2 (TmLibrary bf0f6d455c2b09bcfd9effbf17dac85b45cf9b2b, TmServer 579a6b1458ca325b5c992d56b0c540b751fae626), there is an issue with the change of type checking. When one does not provide a required input to the workflow (n_vertical and n_horizontal in metaconfig in my case), the pipeline fails [as expected]. But then, if I put the correct values in, I can neither save nor submit. uwsgi always contains the following error:

```
2017-07-21 13:59:25 | INFO     | tmserver.api.workflow                    | save workflow description for experiment 1
Traceback (most recent call last):
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/tissuemaps/.local/lib/python2.7/site-packages/flask_jwt/__init__.py", line 177, in decorator
    return fn(*args, **kwargs)
  File "/home/tissuemaps/tmserver/tmserver/util.py", line 135, in wrapped
    return f(*args, **kwargs)
  File "/home/tissuemaps/tmserver/tmserver/util.py", line 240, in wrapped
    return f(*args, **kwargs)
  File "/home/tissuemaps/tmserver/tmserver/api/workflow.py", line 526, in update_workflow_description
    workflow_description = WorkflowDescription(**data['description'])
  File "/home/tissuemaps/tmlibrary/tmlib/workflow/description.py", line 67, in __init__
    self.add_stage(WorkflowStageDescription(self.type, **stage))
  File "<decorator-gen-6>", line 2, in __init__
  File "/home/tissuemaps/tmlibrary/tmlib/utils.py", line 421, in wrapper
    return func(*args, **kwargs)
  File "/home/tissuemaps/tmlibrary/tmlib/workflow/description.py", line 206, in __init__
    batch_args = BatchArgs(**batch_arg_values)
  File "/home/tissuemaps/tmlibrary/tmlib/workflow/args.py", line 311, in __call__
    return ABCMeta.__call__(cls, *args, **kwargs)
  File "/home/tissuemaps/tmlibrary/tmlib/workflow/args.py", line 344, in __init__
    setattr(self, name, kwargs[name])
  File "/home/tissuemaps/tmlibrary/tmlib/workflow/args.py", line 239, in __set__
    % (self.name, self.type.__name__)
TypeError: Argument "n_horizontal" must have type int.
```

**Comment by scottberry** [comment]
I'm also having the same error. This urgently needs fixing. @hackermd @riccardomurri

**Comment by hackermd** [comment]
Should be fixed with TissueMAPS/TmLibrary@b2296de5d9d94aecdfbd3c5ac7af60c82f4b2f8e

**Comment by scottberry** [comment]
Confimed fixed. Thanks @hackermd
